xunit-autolabeler-v2/ast_parser/lib/file_utils.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2020 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import os
+from os import path
+import subprocess
+
+from .constants import REGION_TAG_GREP_ARGS
+
+
+def __getFiles(root_dir, predicate):
+    if path.basename(path.normpath(root_dir)).startswith('.'):
+        # Ignore dot-directories
+        return []
+
+    paths = [path.join(root_dir, p) for p in os.listdir(root_dir)
+             if p != 'lib']
```

Comment:
Can you safely ignore `lib` directory? Actually you have `lib` directory in your module structure.

Reply:
This code is intended to work with code snippet repos (where, to my knowledge, `lib` should be ignored), not the parser repo itself.

Reply:
Sorry, I don't understand why you can ignore `lib` in sample repos. Sample repo can have somewhat complex project and it can have a legitimate `lib` directory.

Reply:
GAE samples [use this folder to vendor dependencies](https://github.com/GoogleCloudPlatform/python-docs-samples/search?l=Python&q=lib).

Dependencies **should not** be processed by the parser, so that's why I'm ignoring this folder.

If you'd prefer, I'm happy to restrict this to the `appengine` subfolder.